build_tools/cmake/iree_run_module_test.cmake

```diff
@@ -0,0 +1,222 @@
+        endif()
+      endforeach(_FLAG)
+    elseif(NOT _FLAGFILE_PATH AND NOT _XFAIL)
+      message(SEND_ERROR "Cannot find the benchmark module for ${_RULE_BENCHMARK_MODULE_SRC}")
+    endif(_FLAGFILE_PATH)
+  endif(_RULE_MODULE_SRC)
+
+  list(APPEND _RULE_RUNNER_ARGS "--module_file=${_SRC}")
+
+  if(_RULE_EXPECTED_OUTPUT)
+    get_filename_component(_OUTPUT_FILE_TYPE "${_RULE_EXPECTED_OUTPUT}" EXT)
+    if(_OUTPUT_FILE_TYPE STREQUAL ".txt")
+      get_filename_component(_OUTPUT_FILE_SRC "${_RULE_EXPECTED_OUTPUT}" REALPATH)
+      # Process the text input to remove the line breaks.
+      file(STRINGS "${_OUTPUT_FILE_SRC}" _EXPECTED_OUTPUT ENCODING UTF-8)
+      string(REPLACE ";" " " _EXPECTED_OUTPUT_STR "${_EXPECTED_OUTPUT}")
+      set(_EXPECTED_OUTPUT_STR "--expected_output=\"${_EXPECTED_OUTPUT_STR}\"")
+      list(APPEND _RULE_RUNNER_ARGS ${_EXPECTED_OUTPUT_STR})
+      list(APPEND _SRC ${_OUTPUT_FILE_SRC})
+    elseif(_OUTPUT_FILE_TYPE STREQUAL ".npy")
+      find_file(_OUTPUT_FILE_SRC_PATH
+        NAME
+          "${_RULE_EXPECTED_OUTPUT}"
+        PATHS
+          "${CMAKE_CURRENT_SOURCE_DIR}"
+          "${CMAKE_CURRENT_BINARY_DIR}"
+          "${IREE_BENCHMARK_SUITE_DIR}"
+        NO_CACHE
+      )
+      # If the expected output npy file is not found (the large file is not
+      # loaded from GCS to `IREE_BENCHMARK_SUITE_DIR` benchmark suite test),
+      # report error.
+      if(NOT _OUTPUT_FILE_SRC_PATH AND NOT _XFAIL)
+        message(SEND_ERROR "${_RULE_EXPECTED_OUTPUT} is not found in\n\
+          ${CMAKE_CURRENT_SOURCE_DIR}\n\
+          ${CMAKE_CURRENT_BINARY_DIR}\n\
+          ${IREE_BENCHMARK_SUITE_DIR}\n\
+          Please check if you need to download it first.")
+      elseif(NOT _OUTPUT_FILE_SRC_PATH AND _XFAIL)
+        list(APPEND _RULE_RUNNER_ARGS "--expected_output=fail")
+      elseif(_OUTPUT_FILE_SRC_PATH)
+        get_filename_component(_OUTPUT_FILE_SRC "${_OUTPUT_FILE_SRC_PATH}" REALPATH)
+        list(APPEND _RULE_RUNNER_ARGS "--expected_output=@${_OUTPUT_FILE_SRC}")
+        list(APPEND _SRC ${_OUTPUT_FILE_SRC})
+      endif()
+    else()  # The expected output is listed in the field.
+      list(APPEND _RULE_RUNNER_ARGS "--expected_output=\"${_RULE_EXPECTED_OUTPUT}\"")
+    endif()
+  endif(_RULE_EXPECTED_OUTPUT)
+
+  if(NOT DEFINED _RULE_DRIVER)
+    return()
+  endif()
+
+  # Dump the flags into a flag file to avoid CMake's naive handling of spaces
+  # in expected output.
+  if(_RULE_RUNNER_ARGS)
+    set(_OUTPUT_FLAGFILE "${CMAKE_CURRENT_BINARY_DIR}/${_RULE_NAME}_flagfile")
+    # Write each argument in a new line.
+    string(REPLACE ";" "\n" _OUTPUT_FLAGS "${_RULE_RUNNER_ARGS}")
+    file(WRITE "${_OUTPUT_FLAGFILE}" "${_OUTPUT_FLAGS}")
+    list(APPEND _SRC "${CMAKE_CURRENT_BINARY_DIR}/${_RULE_NAME}_flagfile")
+  endif()
+
+  # A target specifically for the test. We could combine this with the above,
+  # but we want that one to get pulled into iree_bytecode_module.
```

Comment:
I don't understand what you mean about combining it with the above. Also, this target doesn't seem to depend on anything. I don't think building it will trigger anything

Reply:
Clean up the comment.